ompi/mca/hook/comm_method/hook_comm_method_component.c

```diff
@@ -132,21 +145,28 @@ static int ompi_hook_comm_method_component_register(void)
                                            &mca_hook_comm_method_enable_mpi_finalize);
 
     // User can set the comm_method mca variable too
-    int hook_comm_method = -1;
-    (void) mca_base_var_register("ompi", NULL, NULL, "comm_method",
-                                 "Enable comm_method behavior (1) mpi_init or (2) mpi_finalize",
-                                 MCA_BASE_VAR_TYPE_INT, NULL,
-                                 0, 0,
-                                 OPAL_INFO_LVL_3,
-                                 MCA_BASE_VAR_SCOPE_READONLY,
-                                 &hook_comm_method);
-
-    if( 1 == hook_comm_method ) {
+    mca_base_var_enum_create("ompi_comm_method", mca_hook_comm_method_modes, &new_enum);
+    ret = mca_base_var_register("ompi", NULL, NULL, "comm_method",
+                                "Enable comm_method behavior (1) mpi_init or (2) mpi_finalize",
```

Comment:
Does this help message need to be updated to show the words and/or the 3/all option?

Reply:
Yeah, that's probably a good idea. It's part of the output from the registration, but I can clean up the main description.

Reply:
Fixed in latest push.